src/components/organisms/PortalAddForm/PortalAddForm.tsx

```diff
@@ -0,0 +1,152 @@
+  PortalInfoListItem,
+  ROOM_TAXON,
+  SPACE_TAXON,
+} from "settings";
+
+import { createRoom, createSlug, createVenue_v2 } from "api/admin";
+
+import { PortalInput } from "types/rooms";
+
+import { venueInsideFullUrl } from "utils/url";
+import { buildEmptySpace } from "utils/venue";
+
+import { createPortalSchema } from "forms/createPortalSchema";
+import { createSpaceSchema } from "forms/createSpaceSchema";
+
+import { useSpaceBySlug } from "hooks/spaces/useSpaceBySlug";
+import { useSpaceParams } from "hooks/spaces/useSpaceParams";
+import { useCheckImage } from "hooks/useCheckImage";
+import { useUser } from "hooks/useUser";
+import { useWorldBySlug } from "hooks/worlds/useWorldBySlug";
+
+import { AdminInput } from "components/molecules/AdminInput";
+import { SubmitError } from "components/molecules/SubmitError";
+
+import { ButtonNG } from "components/atoms/ButtonNG";
+
+import "./PortalAddForm.scss";
+
+export interface PortalAddFormProps {
+  item: PortalInfoListItem;
+  onDone: () => void;
+}
+
+export const PortalAddForm: React.FC<PortalAddFormProps> = ({
+  item,
+  onDone,
+}) => {
+  const { description, icon, poster, template, text } = item;
+
+  const { user } = useUser();
+  const { spaceSlug, worldSlug } = useSpaceParams();
+  const { worldId } = useWorldBySlug(worldSlug);
+  const { spaceId } = useSpaceBySlug(spaceSlug);
+
+  const { register, getValues, handleSubmit, errors } = useForm({
+    validationSchema:
+      template === "external" ? createPortalSchema : createSpaceSchema,
+    defaultValues: {
+      roomTitle: "",
+      roomUrl: "",
+      venueName: "",
+      template,
+    },
+  });
+
+  const [
+    { loading: isLoading, error: submitError },
+    addPortal,
+  ] = useAsyncFn(async () => {
+    if (!user || !spaceSlug || !template || !worldId || !spaceId) return;
+
+    const { roomUrl, venueName } = getValues();
+
+    const slug = createSlug(venueName);
+    const url = template === "external" ? roomUrl : venueInsideFullUrl(slug);
+
+    const portalData: PortalInput = {
+      ...DEFAULT_PORTAL_INPUT,
+      title: venueName,
+      image_url: icon,
+      url,
+      template,
+    };
+
+    if (template !== "external") {
+      const venueData = buildEmptySpace(venueName, template);
+      const newSpaceSlug = createSlug(venueName);
+
+      await createVenue_v2(
+        {
+          ...venueData,
+          worldId,
+          parentId: spaceSlug, // @debt this should be parent's spaceId, and if needed unambiguous identifiers, uncomment the following
+          // parentSpaceId: spaceId,
+          // parentSpaceSlug: spaceSlug,
+          logoImageUrl: icon,
+          name: newSpaceSlug, // @debt this should be unfiltered venueName once the possible bugs of mixing slug with id are resolved
+          slug: newSpaceSlug,
+        },
+        user
+      );
+    }
+
+    // @debt this is wrong, for all intents and purposes the venueId a.k.a spaceId is the identifier of the parent space
```

Comment:
I don't think this debt comment is correct. `createRoom` is expecting to receive the ID of the space that owns the portal/room. That's what the code does.

Perhaps the variables could be better named. `fromSpaceId` or similar.

Reply:
The comment isn't about the name, but the value. ID wasn't provided, but a slug, as far as I can remember. This of course might have been copied from the original.

Reply:
removed the comment @ 72db53c90a085bd17a75620de6a456171e6988dc